pkg/crypto/api.go

```diff
@@ -84,6 +88,30 @@ type Crypto interface {
 	// 		signature proof in []byte
 	//		error in case of errors
 	DeriveProof(messages [][]byte, bbsSignature, nonce []byte, revealedIndexes []int, kh interface{}) ([]byte, error)
+
+	// CL Anoncreds methods
```

Comment:
wouldn't it make more sense to create a separate Crytpo interface for CL? eg `type CryptoCL interface{...}`

Reply:
we want these CL methods to be accessible via Crypto service.
or you mean to move all CL methods to CryptoCL and then make Crypto API extends this CL interface?

Reply:
Yes we need CryptoCL interface

Reply:
Not necessarily extend. It depends on how these functions are used. Crypto interface is for general crypto service uses like encrypt, sign, hmac, keywrap, etc. The new CrytoCL interface is independent of Crypto and has specific credentials operations.

If someone wants to provide their own Crypto implementation, should they have to create empty CL functions if they're not used? This is the main question, imho, it's better to keep these 2 services independent.

Reply:
As I understand Crypto API already contains BBS+ specific functions as `SignMulti` and `VerifyMulti`. CL Anoncreds are not different from them in their narrow specifics.
The main point in adding CL methods to Crypto API is to leverage DI af-go provides and be able to use different implementations of crypto - like `tinkcrypto` and `remotecrypto` as well (we're planning to add corresponding support in the one of the next PRs)